examples/water-model/water-model.py

```diff
@@ -330,320 +354,269 @@
-    HOH_equilibrium_angle=HOH_equilibrium_angle,
-    pme_smearing=pme_smearing,
-    pme_mesh_spacing=pme_mesh_spacing,
-    pme_interpolation_nodes=pme_interpolation_nodes,
-    pme_prefactor=pme_prefactor,
-    four_point_model=False,
+from metatensor.torch.atomistic import (
+    MetatensorAtomisticModel,
+    ModelCapabilities,
+    ModelEvaluationOptions,
+    ModelMetadata,
+    ModelOutput,
+    NeighborListOptions,
+    System,
 )
-spc_f_model.eval()
 
-spc_f_model_atomistic_model = MetatensorAtomisticModel(
-    spc_f_model, ModelMetadata(), model_capabilities
+atoms = ase.io.read("data/water_32.pdb")
+system = System(
+    types=torch.from_numpy(atoms.get_atomic_numbers()),
+    positions=torch.from_numpy(atoms.positions),
+    cell=torch.from_numpy(atoms.cell.array),
+    pbc=torch.from_numpy(atoms.pbc),
 )
+nlo = NeighborListOptions(cutoff=6.0, full_list=False, strict=False)
 
-# %%
-#
-# We run the MD simulation in the constant volume/temperature (NVT) ensemble at 300
-# Kelvin, using a Langevin thermostat for integration. First we set the
-# ``atomistic_model`` as the calculator for our system.
-
-water_mta_calculator = MetatensorCalculator(spc_f_model_atomistic_model)
-atoms.calc = water_mta_calculator
-
-# %%
-#
-# To test we can compute the potential energy of the system.
-
-print(f"Potential energy of the system: {atoms.get_potential_energy():.2f} eV")
-
-# %%
-#
-# Set initial velocities according to the Maxwell-Boltzmann distribution at 300 Kelvin.
-
-ase.md.velocitydistribution.MaxwellBoltzmannDistribution(atoms, temperature_K=300)
-
-# %%
-#
-# Set up the Langevin thermostat for the NVT ensemble.
 
-integrator = ase.md.Langevin(
-    atoms,
-    timestep=0.5 * ase.units.fs,
-    temperature_K=300,
-    friction=1 / ase.units.fs,
-)
+from vesin.torch.metatensor import NeighborList
+from metatensor.torch import Labels, TensorBlock, TensorMap
+calculator = NeighborList(nlo, length_unit="Angstrom")
+neighbors = calculator.compute(system)
+system.add_neighbor_list(nlo, neighbors)
 
 # %%
-#
-# Run the Simulation
-# ------------------
-#
-# We run the simulation for 50 steps (:math:`0.5\,\mathrm{fs}`) and collect potential,
-# kinetic, and total energy, as well as temperature and pressure.
-
-n_steps = 50
+# 
+neighbors
+
+# %% 
+#
+
+def get_bonds_angles(positions: torch.Tensor):
+    """ Return the list of bond distances, angles and charge site 
+    positions for the water molecules. These are assumed not to 
+    be "folded", and to be listed in the input as OHHOHHOHH."""
+    
+    o_pos = positions[::3]
+    h1_pos = positions[1::3]
+    h2_pos = positions[2::3]
+
+    oh_dist = torch.concatenate([
+            torch.sqrt(((h1_pos-o_pos)**2).sum(axis=1)),
+            torch.sqrt(((h2_pos-o_pos)**2).sum(axis=1)),
+    ])
+    if oh_dist.max() > 2.0: 
+        raise ValueError("Unphysical O-H bond length. Check that the molecules are entire, and atoms are listed in the expected OHH order.")
```

Comment:
I would not constrain it in this way. The usual definition is as I did bonds should not be larger than half of shortest box length.

Reply:
Given the new LAMMPS heuristics, I'd leave it as this - a O-H above 2Å means sure trouble